base_delivery_carrier_label/models/stock_picking.py

```diff
@@ -85,23 +74,36 @@ def get_shipping_label_values(self, label):
             'file_type': label['file_type'],
         }
 
-    @api.multi
-    def generate_labels(self, package_ids=None):
-        """ Generate the labels.
+    def generate_labels(self):
+        """ Legacy method. Remove me after 12.0
+        """
+        _logger.warning(
+            "Your delivery module depending on your carrier must call "
+            "action_generate_carrier_label() method "
+            "instead of generate_labels()")
+        return self.action_generate_carrier_label()
+
+    def _set_a_default_package(self):
```

Comment:
@bealdav Could you give me a little context to have a package mandatory in this case ? Thanks